### ISSUE TITLE
[BUG]: char arrays not read in properly

### ISSUE THREAD

**Issue by asreimer** [SuperDARN/pydarn#7]
This is just a housekeeping issue post. Make sure we don't lose track of things.

As mentioned in #4, we need to handle char arrays differently when reading dmap arrays. For more details see "Outstanding Issue" in this comment: https://github.com/SuperDARN/pydarn/pull/4#issuecomment-471928380

We can fix this by adding:
    if array_type_fmt == 'c':
        array_value = np.array([ord(x) if len(x) == 1 else 0 for x in array_value])

[right here](https://github.com/SuperDARN/pydarn/blob/31210af672f8a2e5a5bd0df1f6dca70fe386dffb/pydarn/io/dmap.py#L586).

edit: This fix only handles reading in the char arrays. Not sure what happens when we write them out.

**Comment by mts299** [comment]
Okay, we got some decisions to make about this bug. 

Either we treat `char` types as:
1. `int8`, as they are used in the current DMAP SuperDARN files types
2.  `characters` ex) 's'

Currently, `DmapRead` treats them as `int8` and this is how it is implemented in backscatter as well.
Benefits of `int8`:
- keep the current performance 
- less checking 
Cons: 
- not as robust 

The benefit of 'characters':
- robust 
Cons:
- performance hit 
- Potential future complications since python doesn't support `char` types. Instead, it treats `char` as a `string`. 

My current thoughts, if no file types actually have `characters` and just uses `int8` types, then we could enforce in the future to treat `characters` as `strings`.

**Comment by mts299** [comment]
So @asreimer pointed me to `rtypes.h` file in RST and this is how chars are defined:
``` C
    #if WORD_BIT == 32
      typedef char int8;
      typedef unsigned char uint8;
      typedef short int16;
      typedef unsigned short uint16;
      typedef int int32;
      typedef unsigned int uint32;
      typedef long long int64;
      typedef unsigned long long uint64;
    #endif
```

From this I think treating `chars` as `int8` is a safe bet :)

**Comment by asreimer** [comment]
This was resolved in #4
